platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBook.java

```diff
@@ -543,6 +528,49 @@ public Iterator<Address> iterator() {
         return new AddressBookIterator(orderedNodeIds.iterator(), addresses);
     }
 
+    /**
+     * The text form of an address book that appears in config.txt
+     *
+     * @return the string form of the AddressBook that would appear in config.txt
+     */
+    public String toConfigText() {
+        final StringBuilder sb = new StringBuilder();
+
+        for (final Address address : this) {
+            sb.append("  ").append(address.toConfigText()).append(",\n");
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * Parses an address book from text with the assumption
+     *
+     * @param addressBookText the config.txt compatible serialized address book to parse.
+     * @param posToId         a function to determine the address id given the position of the address in the text.
+     * @param isOwnDeterminer a function to determine if the address isOwn property should be true given an
+     *                        InetAddress.
+     * @param memoSource      a function to render memo text given the address id.
+     * @return a parsed AddressBook.
+     * @throws ParseException if any Address throws a ParseException when being parsed.
+     */
+    public static AddressBook parseConfigText(
+            final String addressBookText,
+            final Function<Long, Long> posToId,
+            final Function<InetAddress, Boolean> isOwnDeterminer,
+            final Function<Long, String> memoSource)
+            throws ParseException {
+        Objects.requireNonNull(addressBookText, "The addressBookText must not be null.");
+        AddressBook addressBook = new AddressBook();
+        long pos = 0;
+        for (String addressLine : addressBookText.split("\\r?\\n")) {
```

Comment:
```suggestion
        for (final String addressLine : addressBookText.split("\\r?\\n")) {
```

Reply:
I added final keyword and moved function to AddressBookUtils.java